### DISCUSSION TITLE
generate pictures in CI for README.md with respect to branches

### DISCUSSION THREAD

**Discussion by joric** [yaqwsx/PcbDraw#87] [General]
Subj? See https://github.com/joric/nrfmicro I can upload pictures to the latest release and link to pictures from there but how to create branches without fixing urls in readme? Maybe there's a github proxy that does just that? I don't upload to artifacts because they only live 30 days.

**Comment by yaqwsx** [comment]
I usually build my project every night and produce build artifacts. Then I use https://nightly.link to get the latest link. But I am not sure if that can work with images.

The other solution is to build proper GH pages and link to them. Or you can have a special branch with images and you will have a folder for each branch, where you store the images.

**Comment by joric** [comment]
No, artifacts don't support images, zip files only. Another idea - what if upload pictures to wiki via CI? The issue is that it might get huge because of binary files diffs but it works and it doesn't need setting up a repository and it's practically invisible (very few people really download wikis locally).

There's still an issue with readme links though they won't change if the branch changes, I don't think it's possible to make them relative both to the branch and to the picture in the wiki and I really don't want to fix readme after branching.

Looks like extracting a branch name from readme links like https://github.com/joric/nrfmicro/tree/main and pointing them to a file like https://github.com/joric/nrfmicro/wiki/uploads/nrfmicro-main-back.png is impossible without a server. Sadly even github apps don't give you a server (you need your own hosting). A javascript-based redirector on github pages won't work for embedded images.
